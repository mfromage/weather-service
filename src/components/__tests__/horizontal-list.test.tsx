import { render } from '@testing-library/react-native';
import { Text } from 'react-native';
import HorizontalList from '@/components/horizontal-list';

describe('Horizontal List', () => {
  const setup = () => {
    const testId = 'horizontal-list-test-id';
    const utils = render(
      <HorizontalList testID={testId}>
        <Text>Text 1</Text>
        <Text>Text 2</Text>
      </HorizontalList>,
    );
    const horizontalList = utils.getByTestId(testId);
    return { ...utils, horizontalList };
  };
  it('render with given props', () => {
    const { horizontalList } = setup();
    expect(horizontalList).toBeTruthy();
  });

  it('matches the snapshot', () => {
    const { toJSON } = setup();
    expect(toJSON()).toMatchSnapshot();
  });

  it('contains the children', () => {
    const { horizontalList } = setup();
    expect(horizontalList.children.length).toBe(2);
  });
});
