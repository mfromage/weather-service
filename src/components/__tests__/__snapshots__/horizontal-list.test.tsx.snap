// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Horizontal List matches the snapshot 1`] = `
<RCTScrollView
  horizontal={true}
>
  <View>
    <View
      style={
        {
          "flexDirection": "row",
          "gap": 8,
        }
      }
      testID="horizontal-list-test-id"
    >
      <Text>
        Text 1
      </Text>
      <Text>
        Text 2
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
